### PR TITLE
Add `rustsim` to dependencies; add implementation stub

### PR DESCRIPTION
**Pull request by caufieldjh**
Use rustsim (https://github.com/Knowledge-Graph-Hub/rustsim) for semantic similarity comparisons.
Results should be approximately identical to stock OAK semsim but faster.

This PR is just for including the rustsim dependency and adding the basic functionality.

### CODE REVIEW

**Comment by caufieldjh** [comment]
Test fails for 3.10 (on [5051f19](https://github.com/INCATools/ontology-access-kit/pull/451/commits/5051f19442da7e077d4f5cbf7972839ab154d5d2)) because we don't (currently) publish a 3.10 wheel.

File: pyproject.toml [hunk 1]
```diff
@@ -25,15 +25,15 @@ kgcl-schema = "^0.3.1"
...
-pystow = "^0.5.0"
+pystow = "^0.4.4"
 class-resolver = "^0.3.10"
 ontoportal-client = "0.0.3"
 bioregistry = "^0.5.64"
 prefixmaps = "^0.1.2"
 ols-client = "^0.1.1"
 pandas = "^1.5.1"
 linkml-renderer = "^0.1.2"
-airium = "^0.2.5"
```
Comment (cmungall) [inline pyproject.toml:36]: was there a reason for this?

File: pyproject.toml [hunk 2]
```diff
@@ -25,15 +25,15 @@ kgcl-schema = "^0.3.1"
 funowl = "^0.1.12"
 gilda = {version = "^0.10.1", optional = true}
 kgcl-rdflib = "^0.3.0"
-pystow = "^0.5.0"
+pystow = "^0.4.4"
```
Comment (cmungall) [inline pyproject.toml:28]: ideally we would not do this

Comment (caufieldjh) [reply pyproject.toml]: Reverted in 9e38b39c9106440256fe1ad5407bd3c4187d0338

File: pyproject.toml (hunk 1 above)
Comment (caufieldjh) [reply pyproject.toml:36]: Reverted in 9e38b39c9106440256fe1ad5407bd3c4187d0338

Comment (caufieldjh) [reply pyproject.toml:36]: I don't know why that happened.

**Comment by caufieldjh** [comment]
Windows tests will fail because we don't build a rustsim windows wheel - see https://github.com/Knowledge-Graph-Hub/rustsim/issues/16

**Comment by caufieldjh** [comment]
Move rustsim to Implementations from Interfaces, following design patterns in this project

**Comment by cmungall** [comment]
I assume this is subsumed into #488 now?

**Comment by caufieldjh** [comment]
Yes, let's consider this a redundant PR

**Pull request closed without merging**